### PR TITLE
Snapshot interfaces

### LINKED ISSUE #4616: Update /cdns/{{name}}/snapshot to use data from new interfaces tables
## I'm submitting a ...
- new feature / enhancement request

## Description
The `/cdns/{{name}}/snapshot` endpoint need to be updated to utilize data from the new tables added by whatever PR resolves #4611.

This depends on #4611

## Traffic Control components affected ...
- Traffic Ops client (Go)

## Acceptance Criteria
* `/cdns/{{name}}/snapshot` in API 1.x AND API 2.x:
	* Provides the proper data in representations of cache servers as outlined in the blueprint for the `GET` method
* Any associated unit and/or client/API integration tests all pass.

See the [Additional Server Interfaces blueprint](https://github.com/apache/trafficcontrol/blob/master/blueprints/multi-interface-servers.md#cdnsnamesnapshot) for details.

### LINKED ISSUE #4617: Update /cdns/{{name}}/snapshot/new to use data from the new interfaces tables
## I'm submitting a ...
- new feature / enhancement request

## Description
The `/cdns/{{name}}/snapshot/new` endpoint need to be updated to utilize data from the new tables added by whatever PR resolves #4611.

This depends on #4611

## Traffic Control components affected ...
- Traffic Ops

## Acceptance Criteria
* `/cdns/{{name}}/snapshot/new` in API 1.x AND API 2.x:
	* Provides the proper data in representations of cache servers as outlined in the blueprint for the `GET` method
* Any associated unit and/or client/API integration tests all pass.

See the [Additional Server Interfaces blueprint](https://github.com/apache/trafficcontrol/blob/master/blueprints/multi-interface-servers.md#cdnsnamesnapshotnew) for details.

### PR DESCRIPTION
**Pull request by ocket8888**
## What does this PR (Pull Request) do?
- [x] This PR fixes #4616 and fixes #4617

Updates CDN Snapshot generation to account for multiple server interfaces.

## Which Traffic Control components are affected by this PR?
- Traffic Ops

## What is the best way to verify this PR?
Run the unit and API tests, run CDN-in-a-Box and verify the snapshot is successful.

## The following criteria are ALL met by this PR
- [x] No interface change, so existing tests should be sufficient
- [x] No interface change, so no documentation necessary
- [x] This PR includes an update to CHANGELOG.md
- [x] This PR includes any and all required license headers
- [x] This PR does not include a database migration
- [x] This PR **DOES NOT FIX A SERIOUS SECURITY VULNERABILITY**

### CODE REVIEW

**Comment by rawlinp** [comment]
Unit test job failed with this:
```
unit_1  | === RUN   TestGetMidServers
unit_1  |     TestGetMidServers: servers_test.go:475: getMidServers expected: len(mid) == 1, actual: 0
unit_1  | --- FAIL: TestGetMidServers (0.00s)
```

File: traffic_ops/traffic_ops_golang/dbhelpers/db_helpers.go [hunk 1]
```diff
@@ -519,47 +519,72 @@ WHERE s.id = $1
...
+	       interface,
+	       server
+	FROM ip_address
+	WHERE ip_address.server = ANY ($1)
+	`
+	ipRows, err := tx.Query(q, pq.Array(ids))
+	if err != nil {
+		return nil, err
+	}
+
```
Comment (rawlinp) [inline traffic_ops/traffic_ops_golang/dbhelpers/db_helpers.go:568]: Don't forget to `defer ipRows.Close()`, or even better `defer log.Close(ipRows, "...")

Comment (ocket8888) [reply traffic_ops/traffic_ops_golang/dbhelpers/db_helpers.go:568]: whoops

**Comment by ocket8888** [comment]
I can't reproduce that failure. All unit tests are passing for me.

**Comment by rawlinp** [comment]
Ok, I'll pull this down and test when the CI completes again.

**Comment by shamrickus** [comment]
The unit tests pass for me as well.

**Pull request merged**